lib/Dialect/SV/Transforms/SVExtractTestCode.cpp

```diff
@@ -84,6 +84,33 @@ static void getBackwardSliceSimple(Operation *rootOp,
   backwardSlice.remove(rootOp);
 }
 
+// Reimplemented and simplified from SliceAnalysis to use a worklist rather than
+// recursion and not consider nested regions. Also accepts a clone set
+// representing a backward dataflow slice. As soon as a use is found outside the
+// backward slice, return false. Otherwise, the whole forward slice is contained
+// in the backward slice, and return true.
+static bool getForwardSliceSimple(Operation *rootOp,
+                                  SetVector<Operation *> &opsToClone,
+                                  SetVector<Operation *> &forwardSlice) {
+  SmallPtrSet<Operation *, 32> visited;
+  SmallVector<Operation *> worklist;
+  worklist.push_back(rootOp);
+
+  while (!worklist.empty()) {
+    Operation *op = worklist.pop_back_val();
+    forwardSlice.insert(op);
+    visited.insert(op);
+    for (auto *user : op->getUsers()) {
+      if (!opsToClone.contains(user))
+        return false;
+      if (!visited.contains(user))
+        worklist.push_back(user);
+    }
+  }
```

Comment:
That implementation makes sense to me but probably the following implementation is slightly more efficient. 
```suggestion
  worklist.push_back(rootOp);
  visited.insert(rootOp);

  while (!worklist.empty()) {
    Operation *op = worklist.pop_back_val();
    forwardSlice.insert(op);

    for (auto *user : op->getUsers()) {
      if (!opsToClone.contains(user))
        return false;
      if (visited.insert(op).second)
        worklist.push_back(user);
    }
  }
```